pykafka/balancedconsumer.py

```diff
@@ -259,6 +268,11 @@ def stop(self):
             # facilities for that in ChildrenWatch - as a workaround we check
             # self._running in the watcher callbacks (see further down)
 
+        # Unregister the cleanup handler
+        unregister_cleanup_func(self._cleanup_func, self)
```

Comment:
One nitpick would be that this stores a reference to `self` elsewhere, which would prevent garbage collection. That's a bit academic currently, because we already have other reasons why gc doesn't happen, but this would add one more.